test/integration/api/allocation/idle_cost_negative_test.go

```diff
@@ -0,0 +1,184 @@
+// This test primarily focuses on the __idle__ namespace (when the breakdown is grouped by namespace)
+// idle costs - https://opencost.io/docs/specification#idle-costs by definition shouldn't be negative
+
+// Test Cases
+// Check all Resouce Usage Costs such as RAM, GPU, PersisitentVolume, LoadBalancer, etc
+// Check for different time windows and breakdown
+// Only __idle__ is tested here and not the resources (like ingress-nginx or folding-at-home)
+
+// Pass Criteria
+// No idle cost for any resource type is negative
+
+import (
+	// "fmt"
+	// "time"
+	"testing"
+	"github.com/opencost/opencost-integration-tests/pkg/api"
+)
+
+// Checks relevant cost fields in an AllocationResponseItem for negative values
+func checkNegativeCosts(m api.AllocationResponseItem) (bool, []string){
+
+	isNegative := false // Flag to track if any negative value is found
+	var negativeFields []string // To store names of negative fields
+
+	// Check each field individually
+	if m.CPUCost < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "CPUCost")
+	}
+	if m.GPUCost < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "GPUCost")
+	}
+	if m.RAMCost < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "RAMCost")
+	}
+	if m.NetworkCost < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "NetworkCost")
+	}
+	if m.LoadBalancerCost < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "LoadBalancerCost")
+	}
+
+	// Check Idle-Cost fields
+	if m.CPUCostIdle < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "CPUCostIdle")
+	}
+	if m.GPUCostIdle < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "GPUCostIdle")
+	}
+	if m.RAMCostIdle < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "RAMCostIdle")
+	}
+
+	// Check adjustment cost fields
+	if m.CPUCostAdjustment < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "CPUCostAdjustment")
+	}
+	if m.GPUCostAdjustment < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "GPUCostAdjustment")
+	}
+	if m.NetworkCostAdjustment < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "NetworkCostAdjustment")
+	}
+	if m.LoadBalancerCostAdjustment < 0 {
+		isNegative = true
+		negativeFields = append(negativeFields, "LoadBalancerCostAdjustment")
+	}
+
+	return isNegative, negativeFields
+}
+
+func TestNegativeIdleCosts(t *testing.T) {
+	apiObj := api.NewAPI()
+
+	testCases := []struct {
+		name		string
+		window      string
+		aggregate   string
+		accumulate  string
+		includeidle string
+	}{
+		{
+			name: "Today",
+			window: "today",
+			aggregate: "namespace",
+			accumulate: "false",
+			includeidle: "true",
+		},
+		// { // This test is meant to fail because there is no includeidle field, i.e no __idle__
+		// 	name: "Missing includeIdle",
+		// 	window: "today",
+		// 	aggregate: "namespace",
+		// 	accumulate: "false",
+		// },

```

Comment:
could you then modify this test case to have an expectError field, setting to true here?